fvtest/jitbuildertest/GlobalTest.cpp

```diff
@@ -0,0 +1,149 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#include "JBTestUtil.hpp"
+
+static int64_t globalInt64 = 0;
+
+static int32_t globalInt32 = 0;
+
+static int16_t globalInt16 = 0;
+
+static int8_t globalInt8 = 0;
+
+DEFINE_BUILDER(TestInt64Global,
+               Int64,
+               PARAM("param", Int64))
+   {
+   DefineGlobal("int64global",Int64,(void*)(&globalInt64));
+
+   OMR::JitBuilder::IlValue *param = Load("param");
```

Comment:
These tests would be better if they picked up a global value set outside the function. Otherwise, all you've tested is that the compiled method can store and reload a "global" value. But you haven't tested that the location matches the variable used by external code. @Leonardo2718 may have thoughts on how to construct such tests...

If we can't write tests that verify it, then you should, at a minimum, change the statically initialized value for each of the global variables to "something", then have the compiled code load that global variable, add something to it, and return the sum for validation.

Reply:
I have added load-only tests in https://github.com/eclipse/omr/pull/4774/commits/da04cd989c8b031d3444489d397f6297da254535. Are they appropriate for verifying an external variable's value?

Reply:
is it not possible to write a test that verifies the value of those extern variables?

thinking about it a bit more (apologies for not including this comment originally), it would be nice to have the test store to the global variable too (even if we can't yet verify that it changes the value...). At least it would verify the compiler doesn't barf on generating the Store to such a variable and the runtime code doesn't contain something trivially broken.